api_app/script_analyzers/observable_analyzers/stratosphere.py

```diff
@@ -0,0 +1,140 @@
+            "repeated_attacker": False,
+            "repeated_attacker_rating": "_",
+        }
+
+        self.check_dataset_status()
+
+        with open(db_loc0, "r") as f:
+            db = f.read()
+        db_list = db.split("\n")
+        count = 0
+        for ip_tuple in db_list:
+            if count < 2:
+                count += 1
+                continue
+            else:
+                if ip in ip_tuple:
+                    ip_rating = ((ip_tuple.split(","))[2]).strip()
+                    result["last24hrs"] = True
+                    result["last24hrs_rating"] = ip_rating
+
+        with open(db_loc1, "r") as f:
+            db = f.read()
+        db_list = db.split("\n")
+        count = 0
+        for ip_tuple in db_list:
+            if count < 2:
+                count += 1
+                continue
+            else:
+                if ip in ip_tuple:
+                    ip_rating = ((ip_tuple.split(","))[2]).strip()
+                    result["new_attacker"] = True
+                    result["new_attacker_rating"] = ip_rating
+
+        with open(db_loc2, "r") as f:
+            db = f.read()
+        db_list = db.split("\n")
+        count = 0
+        for ip_tuple in db_list:
+            if count < 2:
+                count += 1
+                continue
+            else:
+                if ip in ip_tuple:
+                    ip_rating = ((ip_tuple.split(","))[2]).strip()
+                    result["repeated_attacker"] = True
+                    result["repeated_attacker_rating"] = ip_rating
+
+        return result
+
+    def updater(self):
+        try:
+            logger.info("starting download of dataset from stratosphere")
+
+            base_url = "https://mcfp.felk.cvut.cz"
+            mid_url = "/publicDatasets/CTU-AIPP-BlackList/Todays-Blacklists/"
+            url0 = base_url + mid_url + "AIP_blacklist_for_IPs_seen_last_24_hours.csv"
+            priority_url = "AIP_historical_blacklist_prioritized_by_"
+            url1 = base_url + mid_url + priority_url + "newest_attackers.csv"
+            url2 = base_url + mid_url + priority_url + "repeated_attackers.csv"
+
+            p = requests.get(url0, verify=False)
+            p.raise_for_status()
+
+            q = requests.get(url1, verify=False)
+            q.raise_for_status()
+
+            r = requests.get(url2, verify=False)
+            r.raise_for_status()
+
+            with open(db_loc0, "w") as f:
+                f.write(p.content.decode())
+
+            with open(db_loc1, "w") as f:
+                f.write(q.content.decode())
+
+            with open(db_loc2, "w") as f:
+                f.write(r.content.decode())
+
+            if not os.path.exists(db_loc0 or db_loc1 or db_loc2):
+                raise AnalyzerRunException("failed extraction of stratosphere dataset")
+
+            logger.info("ended download of dataset from stratosphere")
+
+        except Exception as e:
+            traceback.print_exc()
```

Comment:
this print should be used only for debugging purposes

Reply:
I have changed it to print only during debugging.